src/TraceEvent/ETWTraceEventSource.cs

```diff
@@ -879,7 +879,7 @@ protected override void Dispose(bool disposing)
                 {
                     foreach (TraceEventNativeMethods.SafeTraceHandle handle in handles)
                     {
-                        handle.Dispose();
+                        handle?.Dispose();
```

Comment:
Should you also check if the handle is valid? Is this a standard SafeHandle? If so, calling Dispose on an invalid handle can throw.